### ISSUE TITLE
Herbalism is working partly

### ISSUE THREAD

**Issue by InfurnoHD** [mcMMO-Dev/mcMMO#3207]
So herbalism skill increases if you harvest grass, netherwarts, sugar canes etc.

But when you try to harvest wheat or beats and probably some more, you dont get any exp.

I've tried resetting config files and reinstalling.

Any suggestions?

**Comment by pvprangergod** [comment]
anything i grow does not give xp

**Comment by InfurnoHD** [comment]
I get exp from sugar canes and netherwarts. 

Potato and carrots? Idk

**Comment by Zefferis** [comment]
I've got the same problem, a temporary fix is to go into your world folders and renaming your mcmmo regions folders and restarting the server. It'll generate new mcmmo files and you can get herbalism again, I'm unsure of the direct cause of the issue though :(

**Comment by pvprangergod** [comment]
yup a new world has fixed it for me, any idea what those region folders contain? hope they are not important if i have to remove them later to fix it again

**Comment by Zefferis** [comment]
Oh lordy I hope not, I've deleted that file so many times now just allow herbalism to work, I'm just thankful there's a temp work around.  Gonna do some testing later to see what's causing the problem.

**Comment by pvprangergod** [comment]
ive had the problem again and a new world wont fix it :/ any ideas

**Comment by Zefferis** [comment]
Alright. So I have a plugin called CreeperHeal, apparently it overrideed blocks? I disabled that feature and mcmmo herbalism was working again; do you have any block altering plugins?

**Comment by pvprangergod** [comment]
i removed everything except mcmmo and its not working :(

**Comment by Zefferis** [comment]
When you did that; your testing was with new mcmmo region files right? (other ones were either renamed or deleted to make way for the new to-be-generated regions files)

**Comment by pvprangergod** [comment]
i removed all the worlds and plugins except mcmmo.jar

**Comment by Seeroh** [comment]
I renamed the regions file, and that didn't seem to fix the issue on my end. My server is still on 1.11.2, and has this issue as well, with June 15th's build (191).

**Comment by kitokofox** [comment]
I have to confirm this is an issue.

Some of the players on a server I am on have slowly started isolating the problem, and it seems to stem from player-placed blocks. Harvesting wheat from generated tilled dirt gives MCMMO XP (In an NPC village, for example)

**Comment by Zefferis** [comment]
I thought I had fixed the problem by modifying my creeper heal plugin to stop overridding blocks but it seems the issue persists even with the latest build of mcMMO.

Hmm. However it appears that certain player tilled blocks still give herbalism exp exp and it isn't strickly chunk by chunk (blocks within a chunk can give or not give herbalism exp)

**Comment by Seeroh** [comment]
https://github.com/mcMMO-Dev/mcMMO/commit/15436b44da9d8ad42abf486c5e6795f5424af15e

I think it may have been something in this commit that broke Herbalism.

**Comment by Zefferis** [comment]
Anyone making progress with the issue?

**Comment by Metisuneatn** [comment]
Having this as well with pretty much any crop

**Comment by t00thpick1** [comment]
Looking into it now.

**Comment by Momshroom** [comment]
Thank you.
I just tried build 195.  Still not getting herbalism from carrots, potatoes or wheat.  
Beats, sugarcane, wart and cactus DID give mcmmo xp.   
Cocoa did on my very old farm, but on a staff member's newer one they did not.

**Comment by Momshroom** [comment]
Thank you! Build 196 fixed the xp problem, BUT broke green terra giving full replant on at least wheat.  I haven't tested with other crops.

**Comment by Momshroom** [comment]
This issue has been back for a while. It's also referenced in one way or another in issues:  #3303 #3284 #3154 #2851  and possibly some I missed.

 I've just done a thorough test of all the crops I can think of on Spigot 1.12 with the latest dev build(#5 on the new site).
   
Carrots, potatoes and wart give NO xp.  They only  partially replant with green terra.  If you have high enough herbalism for full replant, they DO fully replant when NOT using green terra.

Wheat gives no xp and does not fully replant, with or without green terra

Cane, cactus, lily pads, red mushrooms, and brown mushrooms give xp as expected.

Old cocoa beans give xp, but do not replant.  If you replant them manually in the same location, and let them grow, they no longer give xp.

Beets give no xp, but DO replant fully with green terra AND with high herbalism and no green terra.

I hope this helps narrow down the issues.

Thank you very much for your time!
(And, yes, all of these are configured to give xp and all used to-- our server's been up a long time.)

**Comment by Glitchfinder** [comment]
So I actually broke down the cause of this over in #3311 a while back. It's an oversight caused by a backend change a while back that made an irrelevant block tag relevant. Specifically, mcMMO now pays attention to crops being tagged as player modified, tags them as player modified when planting, but never actually *un*-tags them. Notably, I just checked out #3384, and it would not address the issue. Sadly, I only ever really have a few minutes at a time to poke at this tracker, so I don't actually have the ability to fix anything for the foreseeable future.
